src/vmspawn/vmspawn.c

```
@@ -0,0 +1,407 @@
+
+                /* These aren't always available so don't raise an error if they fail */
+                (void) copy_xattr(source_fd, NULL, target_fd, NULL, 0);
+                (void) copy_access(source_fd, target_fd);
+                (void) copy_times(source_fd, target_fd, 0);
+
+                r = strv_extend_strv(&cmdline, STRV_MAKE(
+                        "-global", "ICH9-LPC.disable_s3=1",
+                        "-global", "driver=cfi.pflash01,property=secure,value=on",
+                        "-drive"
+                ), false);
+                if (r < 0)
+                        return log_oom();
+
+                r = strv_extendf(&cmdline, "file=%s,if=pflash,format=raw", ovmf_vars_to);
+                if (r < 0)
+                        return log_oom();
+        }
+
+        r = strv_extend(&cmdline, "-drive");
+        if (r < 0)
+                return log_oom();
+
+        r = strv_extendf(&cmdline, "if=none,id=mkosi,file=%s,format=raw", arg_image);
+        if (r < 0)
+                return log_oom();
+
+        r = strv_extend_strv(&cmdline, STRV_MAKE(
+                "-device", "virtio-scsi-pci,id=scsi",
+                "-device", "scsi-hd,drive=mkosi,bootindex=1"
+        ), false);
+        if (r < 0)
+                return log_oom();
+
+        r = strv_extend_strv(&cmdline, arg_parameters, false);
+        if (r < 0)
+                return log_oom();
+
+        pid_t child_pid;
+        r = safe_fork(qemu_binary, 0, &child_pid);
+        if (r == 0) {
+                /* set TERM and LANG if they are missing */
+                if (setenv("TERM", "vt220", 0) < 0)
+                        return log_oom();
+
+                if (setenv("LANG", "C.UTF-8", 0) < 0)
+                        return log_oom();
+
+                execve(qemu_binary, cmdline, environ);
+                log_error_errno(errno, "failed to execve %s: %m", qemu_binary);
+                _exit(r);
```

`_exit(EXIT_FAILURE)` actually. `r` is zero at this point, thus you'd hide the failure here.

also: failed → Failed

---

Fixed in 9c63066737.